django/contrib/postgres/fields/jsonb.py

```diff
@@ -118,6 +118,24 @@ class KeyTextTransform(KeyTransform):
     _output_field = TextField()
 
 
+class CoerceToText(Transform):
```

Comment:
Hm, yeah, these are good pointers. The problem I had that initially made me add it was to make sure the value returned in postgres was fully orderable, as just using the normal `KeyTransform` was causing errors with comparitor operations being undefined. That doesn't really make sense to me, as I thought that was one of the things JSONB was supposed to solve.

Does it make sense to have lookups that convert various things that could be in JSONFields to other Django primitive fields, so we can chain the lookups?

Splitting the PR out makes sense, I've been splitting the `setup_joins` stuff out already, so can do this one too.